insitu/views/action_logging/base_logging.py

```diff
@@ -0,0 +1,125 @@
+import csv
+import json
+import sys
+
+from copernicus.settings import LOGGING_CSV_FILENAME
+from datetime import datetime
+
+
+class BaseLoggingView:
+
+    target_type = ''
+    extra = ''
+
+    def get_object_id(self):
+        return ''
+
+    def log_action(self, request, action, id=''):
+        if 'test' in sys.argv:
+            from copernicus.testsettings import LOGGING_CSV_FILENAME
+        with open(LOGGING_CSV_FILENAME, 'a') as csvfile:
+            spamwriter = csv.writer(csvfile, delimiter=',')
+            row = ",".join([
+                datetime.now().strftime('%d %B %Y %H:%M'),
+                request.user.username,
+                action,
+                self.target_type,
+                str(id),
+                self.extra
+            ])
+            spamwriter.writerow(row.split(','))
+
+
+class GetMethodLoggingView(BaseLoggingView):
+    get_action = None
+
+    def get(self, request, *args, **kwargs):
+        id = self.get_object_id()
+        self.log_action(request, self.get_action, id)
+        return super().get(request, *args, **kwargs)
+
+
+class PostMethodLoggingView(BaseLoggingView):
+    post_action = None
+
+    def post(self, request, *args, **kwargs):
+        response = super().post(request, *args, **kwargs)
+        self.extra = json.dumps(self.get_form().errors)
+        if self.extra:
+            self.post_action = 'tried to create'
```

Comment:
I would have another post_action_failed attribute and overwrite it in the views.